File: app/src/main/java/com/nimtego/plectrum/domain/interactor/InformationSongInteractor.java
package com.nimtego.plectrum.domain.interactor;

import com.nimtego.plectrum.presentation.information_view.song.model.SongDetailsModelK;

import dagger.internal.Preconditions;
import io.reactivex.Observable;

@Deprecated
public class InformationSongInteractor
        extends BaseInteractor<SongDetailsModelK, InformationSongInteractor.Params> {
    @Override
    protected Observable<SongDetailsModelK> buildUseCaseObservable(Params params) {
        Preconditions.checkNotNull(params);
        return repository.songDeteil(params.request);
    }

    public static final class Params {

        private final String request;

        private Params(String request) {
            this.request = request;
        }

        public static Params forRequest(String request) {
            return new Params(request);
        }
    }
}

File: app/src/main/java/com/nimtego/plectrum/domain/interactor/InformationSongInteractor.kt
package com.nimtego.plectrum.domain.interactor

import com.nimtego.plectrum.presentation.mvp.view_model.information_view.SongDetailsModel
import io.reactivex.Observable

class InformationSongInteractor : BaseInteractor<SongDetailsModel, InformationSongInteractor.Params>() {
    override fun buildUseCaseObservable(params: Params): Observable<SongDetailsModel> {
        return repository.songDeteil(params.request)
    }

    class Params private constructor(val request: String) {
        companion object {

            fun forRequest(request: String): Params {
                return Params(request)
            }
        }
    }
}